services/cargohold/test/integration/API/V3.hs

```diff
@@ -76,45 +76,55 @@ tests s = testGroup "v3"
 
 testSimpleRoundtrip :: TestSignature ()
 testSimpleRoundtrip c = do
-    uid <- liftIO $ Id <$> nextRandom
-    uid2 <- liftIO $ Id <$> nextRandom
-
-    -- Initial upload
-    let sets = V3.defAssetSettings
-    let bdy = (applicationText, "Hello World")
-    r1 <- uploadSimple (c . path "/assets/v3") uid sets bdy <!!
-        const 201 === statusCode
-
-    let      loc = decodeHeader "Location" r1 :: ByteString
-    let Just ast = decodeBody r1 :: Maybe V3.Asset
-    let Just tok = view V3.assetToken ast
+    let def  = V3.defAssetSettings
+    let rets = [ V3.AssetVolatile
+               , V3.AssetExpiring
+               , V3.AssetEternalInfrequentAccess
```

Comment:
make `AssetRetention` `deriving (Bounded, Enum)` and define `rets = [minBound ..]`?

Reply:
Good idea https://github.com/wireapp/wire-server/pull/252/commits/dc3e9a8af5250c0d045e96a31aa23c255b4e01a3